examples/custom_potential.jl

```diff
@@ -7,44 +7,37 @@
 using DFTK
 using LinearAlgebra
 
-# First, we define a new element which represents a nucleus generating a custom
-# potential
-struct ElementCustomPotential <: DFTK.Element
-    pot_real::Function      # Real potential
-    pot_fourier::Function   # Fourier potential
+# First, we define a new element which represents a nucleus generating
+# a Gaussian potential.
+struct ElementGaussian <: DFTK.Element
+    α  # Prefactor
+    L  # Width of the Gaussian nucleus
 end
 
-# We need to extend two methods to access the real and Fourier forms of
-# the potential during the computations performed by DFTK
-function DFTK.local_potential_fourier(el::ElementCustomPotential, q::Real)
-    return el.pot_fourier(q)
+# We extend the two methods providing access to the real and Fourier
+# representation of the potential to DFTK.
+
+function DFTK.local_potential_fourier(el::ElementGaussian, q::Real)
+    el.α * exp(- (q * el.L)^2 / 4)
 end
-function DFTK.local_potential_real(el::ElementCustomPotential, r::Real)
-    return el.pot_real(r)
+function DFTK.local_potential_real(el::ElementGaussian, r::Real)
+    el.α * exp(- (r / el.L)^2)
 end
 
 # We set up the lattice. For a 1D case we supply two zero lattice vectors
 a = 10
 lattice = a .* [[1 0 0.]; [0 0 0]; [0 0 0]];
 
 # In this example, we want to generate two Gaussian potentials generated by
-# two nuclei localized at positions ``x_1`` and ``x_2``, that are expressed in
+# two "nuclei" localized at positions ``x_1`` and ``x_2``, that are expressed in
 # ``[0,1)`` in fractional coordinates. ``|x_1 - x_2|`` should be different from
 # ``0.5`` to break symmetry and get nonzero forces.
 x1 = 0.2
-x2 = 0.8;
-# We define the width of the Gaussian potential generated by one nucleus
-L = 0.5;
-# We set the potential in its real and Fourier forms
-pot_real(x) = exp(-(x/L)^2)
-pot_fourier(q::T) where {T <: Real} = exp(- (q*L)^2 / 4);
-# And finally we build the elements and set their positions in the `atoms`
-# array. Note that in this example `pot_real` is not required as all applications
-# of local potentials are done in the Fourier space.
-nucleus = ElementCustomPotential(pot_real, pot_fourier)
-atoms = [nucleus => [x1*[1,0,0], x2*[1,0,0]]];
+x2 = 0.8
+nucleus = ElementGaussian(1.0, 0.5)
```

Comment:
might as well give them a negative potential to get bound states

Reply:
For GPE?

Reply:
Yeah, sure, this GPE is not very different from the linear model. The current examples gets scattering states (which is fine in a finite box), but would be easier to see with bound states.